service/src/main/java/bio/terra/workspace/service/resource/controlled/PrivateResourceState.java

```diff
@@ -0,0 +1,59 @@
+package bio.terra.workspace.service.resource.controlled;
+
+import bio.terra.common.exception.MissingRequiredFieldException;
+import bio.terra.common.exception.SerializationException;
+import bio.terra.workspace.common.exception.InternalLogicException;
+import bio.terra.workspace.generated.model.ApiPrivateResourceState;
+import org.apache.commons.lang3.StringUtils;
+
+public enum PrivateResourceState {
+  ABANDONED("ABANDONED", ApiPrivateResourceState.ABANDONED),
+  ACTIVE("ACTIVE", ApiPrivateResourceState.ACTIVE),
+  INITIALIZING("INITIALIZING", ApiPrivateResourceState.INITIALIZING),
+  LEGACY("LEGACY", ApiPrivateResourceState.LEGACY);
```

Comment:
Added some comments to clarify. You're right that the current setup is inconsistent since the migration sets shared resources to `LEGACY` but they'd be null otherwise - switched to initializing everything to `NOT_APPLICABLE`, and using that for shared resources instead of null.

Setting existing private resources to `ACTIVE` at migration time won't help here - workspaces created before this change won't have the `manager` Sam role defined, so there's no way for WSM to know whether users are still in the workspace or not. That was my original intent for the `LEGACY` role so we could avoid polling (and getting 403) repeatedly.

I'm planning to manually clean most of those up (or read them from the DB and ask other people to clean them up) so in practice there shouldn't be many loophole cases.